File: python/robot-name/robot_name.py
import random
import string

class Robot(object):

    def __init__(self):
        self.usedName = []
        self.name = self.generateName()

    def generateName(self):
        # create name
        name = random.choice(string.ascii_uppercase) + random.choice(string.ascii_uppercase) + str(random.randint(0, 9)) + str(random.randint(0, 9)) + str(random.randint(0, 9))

        if self.verification(name) == True:
            name = self.generateName()

        self.usedName.append(name)
        return name


    # verify that the name is unique
    def verification(self, name):
        for i in range(len(self.usedName)):
            if name == self.usedName[i]:
                return True
        return False



    def reset(self):
        self.name = self.generateName()

File: cpp/robot-name/robot_name.cpp
#include "robot_name.h"

namespace robot_name {

std::string robot::generate_name() {
  std::string robot_id;

  for (int i = 0; i < 2; i++) {
    robot_id += std::toupper('a' + rand() % 26);
  }

  for (int i = 0; i < 3; i++) {
    robot_id += std::to_string(std::rand() % 10);
  }

  return robot_id;
}

robot::robot() {
  std::string new_id = this->generate_name();
  this->id = new_id;
}

std::string robot::name() const {
  return this->id;
}

void robot::reset() {
  std::string old_id = this->id;

  while (old_id == this->id) {
    this->id = this->generate_name();
  }
}

}  // namespace robot_name
